src/package.ts

```diff
@@ -9,6 +9,7 @@ import { exec } from 'shelljs';
 import { fs, Logger } from '@salesforce/core';
 import { AsyncOptionalCreatable } from '@salesforce/kit';
 import { AnyJson, get } from '@salesforce/ts-types';
+import { ChangedPackageVersions } from './commands/npm/dependencies/pin';
```

Comment:
I think this type should be defined in this file. Otherwise, you have a circular dependency.